src/registry.ts

```diff
@@ -106,19 +119,30 @@ export async function getEntry(
       // Transform the response to add quotes, since uint64 cannot be accurately
       // read by JS so the revision needs to be parsed as a string.
       transformResponse: function (data: string) {
+        if (data === undefined) {
+          return {};
+        }
         // Change the revision value from a JSON integer to a string.
         data = data.replace(regexRevisionNoQuotes, '"revision":"$1"');
         // Convert the JSON data to an object.
         return JSON.parse(data);
       },
     });
   } catch (err: unknown) {
-    // unfortunately axios rejects anything that's not >= 200 and < 300
-    return { entry: null, signature: null };
+    // @ts-expect-error TS complains about err.response
+    if (err.response.status === 404) {
+      return { entry: null, signature: null };
+    }
+    // @ts-expect-error TS complains about err.response
+    throw new Error(err.response.data.message);
   }
 
-  if (response.status !== 200) {
-    return { entry: null, signature: null };
+  if (
+    typeof response.data.data != "string" ||
+    typeof response.data.revision != "string" ||
+    typeof response.data.signature != "string"
+  ) {
+    throw new Error("Did not get a complete entry response");
```

Comment:
Is this even possible to generate? Would this error tell user or developer how to proceed?

Reply:
It's better than `type error: data.revision undefined` or something like that. I'll try to elaborate though